app/components/StartStreamingButton.vue.ts

```diff
@@ -97,6 +89,38 @@ export default class StartStreamingButton extends Vue {
     return this.streamingService.state.streamingStatus;
   }
 
+  shouldShowGoLiveWindow() {
+    if (!this.userService.isLoggedIn()) return false;
+
+    if (this.isTwitch) {
+      // For Twitch, we can show the Go Live window even with protected mode off
+      // This is mainly for legacy reasons.
+      return this.customizationService.state.updateStreamInfoOnLive;
+    }
+
+    if (this.isMixer) {
+      return (
+        this.streamSettingsService.protectedModeEnabled &&
+        this.customizationService.state.updateStreamInfoOnLive &&
+        this.streamSettingsService.isSafeToModifyStreamKey()
+      );
+    }
+
+    if (this.isFacebook) {
+      return (
+        this.streamSettingsService.protectedModeEnabled &&
+        this.streamSettingsService.isSafeToModifyStreamKey()
+      );
+    }
+
+    if (this.isYoutube) {
+      return (
+        this.streamSettingsService.protectedModeEnabled &&
+        this.streamSettingsService.isSafeToModifyStreamKey()
+      );
+    }
```

Comment:
this seems a little like unnecessary repetition here-- are you just futureproofing?

Reply:
This was mainly for my own sanity when writing it.  This was originally a very large boolean expression.  For me, the easiest way to verify this was correct was to break it out by what I expected to happen for each platform.  It's now far from a normalized boolean expression, but it was far easier for me to verify the logic was correct, and hopefully far easier to read too.

Reply:
makes sense :+1: for the sake of clarity i think its fine